tequilapi/endpoints/connection_test.go

```diff
@@ -0,0 +1,171 @@
+	fm.disconnectCount += 1
+	return fm.onDisconnectReturn
+}
+
+func (fm *fakeManager) Wait() error {
+	return nil
+}
+
+func TestNotConnectedStateIsReturnedWhenNoConnection(t *testing.T) {
+	var fakeManager = fakeManager{}
+	fakeManager.onStatusReturn = client_connection.ConnectionStatus{
+		State:     client_connection.NOT_CONNECTED,
+		SessionId: "",
+	}
+
+	connEndpoint := NewConnectionEndpoint(&fakeManager)
+	req, err := http.NewRequest(http.MethodGet, "/connection", nil)
+	assert.Nil(t, err)
+	resp := httptest.NewRecorder()
+
+	connEndpoint.Status(resp, req, httprouter.Params{})
+
+	assert.Equal(t, http.StatusOK, resp.Code)
+	assert.JSONEq(
+		t,
+		`{
+			"status" : "NOT_CONNECTED"
+		}`,
+		resp.Body.String())
+}
+
+func TestConnectedStateAndSessionIdIsReturnedWhenIsConnection(t *testing.T) {
+	var fakeManager = fakeManager{}
+	fakeManager.onStatusReturn = client_connection.ConnectionStatus{
+		State:     client_connection.CONNECTED,
+		SessionId: "My-super-session",
+	}
+
+	connEndpoint := NewConnectionEndpoint(&fakeManager)
+	req, err := http.NewRequest(http.MethodGet, "/connection", nil)
+	assert.Nil(t, err)
+	resp := httptest.NewRecorder()
+
+	connEndpoint.Status(resp, req, httprouter.Params{})
+
+	assert.Equal(t, http.StatusOK, resp.Code)
+	assert.JSONEq(
+		t,
+		`{
+			"status" : "CONNECTED",
+			"sessionId" : "My-super-session"
+		}`,
+		resp.Body.String())
+
+}
+
+func TestPutReturns400ErrorIfRequestBodyIsNotJson(t *testing.T) {
+	fakeManager := fakeManager{}
+
+	connEndpoint := NewConnectionEndpoint(&fakeManager)
+	req, err := http.NewRequest(http.MethodPut, "/connection", bytes.NewBufferString("a"))
+	assert.Nil(t, err)
+	resp := httptest.NewRecorder()
+
+	connEndpoint.Create(resp, req, httprouter.Params{})
+
+	assert.Equal(t, http.StatusBadRequest, resp.Code)
+
+	assert.JSONEq(
+		t,
+		`{
+			"message" : "invalid character 'a' looking for beginning of value"
+		}`,
+		resp.Body.String())
+}
+
+func TestPutReturns422ErrorIfRequestBodyIsMissingFieldValues(t *testing.T) {
+	fakeManager := fakeManager{}
+
+	connEndpoint := NewConnectionEndpoint(&fakeManager)
+	req, err := http.NewRequest(http.MethodPut, "/connection", bytes.NewBufferString("{}"))
+	assert.Nil(t, err)
+	resp := httptest.NewRecorder()
+
+	connEndpoint.Create(resp, req, httprouter.Params{})
+
+	assert.Equal(t, http.StatusUnprocessableEntity, resp.Code)
+
+	assert.JSONEq(
+		t,
+		`{
+			"message" : "validation_error",
+			"errors" : {
+				"identity" : [ { "code" : "required" , "message" : "Field is required" } ],
+				"nodeKey" : [ {"code" : "required" , "message" : "Field is required" } ]
+			}
+		}`, resp.Body.String())
+}
+
+func TestPutWithValidBodyCreatesConnection(t *testing.T) {
+	fakeManager := fakeManager{}
+
+	connEndpoint := NewConnectionEndpoint(&fakeManager)
+	req, err := http.NewRequest(
+		http.MethodPut,
+		"/connection",
+		bytes.NewBufferString(
+			`{
+				"identity" : "my-identity",
+				"nodeKey" : "required-node"
+			}`))
```

Comment:
Argument alignment seems to be messed up within most tests. Shouldn't the trailing ")" be on a separate line?

Reply:
It's not java, end of line has implicit ; which breaks golang code :)

Reply:
It works when you do like ), and ) on separate line. Thanks @Waldz  for pointing out.

Reply:
It was @ignasbernotas +1 who pointed out